diktat-static-analysis/save-cloud
# Support configuration overrides for specific servers

### What's done:
* Add example of docker-compose.override.yaml
* Add settings for optional spring properties, mounted from host
* Fix reading of blank `TestRepoList`
* Update docs is `save-deploy/README.md`
* Removed specific overrides for apt proxy, add generic option for additional `apt-get` flags
* Add separate gradle task `buildAndDeployDockerStack`, `deployDockerStack` now tries to get images from the registry by default
* Remove `mdkir`s for mount destinations inside the container, because they are not required
* Add 5 attempts to restart preprocessor if it crashes
* Fix `getHostIp` method in `LinuxUtils`: passing `host.docker.internal` into build command in orchestrator should work correctly now

### TODO:
* Customizing agent properties from orchestrator - should be implemented in a separate issue

Closes #300
